pkg/kubelet/dockertools/docker.go

```diff
@@ -188,7 +192,14 @@ func (p dockerPuller) Pull(image string, secrets []api.Secret) error {
 
 	var pullErrs []error
 	for _, currentCreds := range creds {
-		err := p.client.PullImage(opts, currentCreds)
+		// RegistryAuth is the base64 encoded credentials for the registry
+		base64Auth, err := base64EncodeAuth(currentCreds)
+		if err != nil {
+			return err
```

Comment:
In fact, I suddenly found that it would be cleaner to move `auth` as a separate argument in `PullImage`, and still do the base64 encoding inside the kube_docker_client.go.
However, you have done so much, and the current implementation also works well. I don't think I should bother you with this change. :)